src/main/webapp/WEB-INF/static/js/trialmanager/germplasm.js

```diff
@@ -1,214 +1,753 @@
+					var columns = columnsData.slice();
+					columns.unshift({
+						alias: "",
+						factor: true,
+						name: "CHECK",
+						termId: -6,
+					});
+					return columns;
+				}
+
+				function mapColumns(columnsData) {
+					$scope.columnDataByInputTermId = {};
+
+					var columns = [],
+						columnsDef = [];
+
+					angular.forEach(columnsData, function (columnData, index) {
+						if (columnData.possibleValues) {
+							columnData.possibleValuesByName = {};
+							columnData.possibleValuesById = {};
+							angular.forEach(columnData.possibleValues, function (possibleValue) {
+								// so we can use "Please Choose"=empty value
+								possibleValue.displayValue = possibleValue.name;
+								// convenience map to avoid looping later
+								columnData.possibleValuesByName[possibleValue.name] = possibleValue;
+								columnData.possibleValuesById[possibleValue.id] = possibleValue;
+							});
+							// waiting for https://github.com/angular-ui/ui-select/issues/152
+							columnData.possibleValues.unshift({name: '', displayValue: 'Please Choose', displayDescription: 'Please Choose'});
+						}
+						columnData.index = index;
+
+						columns.push({
+							title: columnData.alias,
+							name: columnData.alias,
+							data: function (row) {
+								return row.properties[columnData.termId];
+							},
+							defaultContent: '',
+							columnData: columnData
+						});
+
+						// CheckBox Column
+						if (columnData.index === 0) {
+							columnsDef.push({
+								targets: columns.length - 1,
+								orderable: false,
+								createdCell: function (td, cellData, rowData, rowIndex, colIndex) {
+									$(td).append($compile('<span><input type="checkbox" ng-click="toggleSelect(' + rowData.entryId + ')"></span>')($scope));
+								}
+							});
+						} else if (columnData.termId === 8240 || columnData.termId === 8250) {
+							// GID or DESIGNATION
+							columnsDef.push({
+								targets: columns.length - 1,
+								orderable: false,
+								render: function (data, type, full, meta) {
+									return '<a class="gid-link" href="javascript: void(0)" ' +
+										'onclick="openGermplasmDetailsPopopWithGidAndDesig(\'' +
+										full.gid + '\',\'' + full.designation + '\')">' + EscapeHTML.escape(data.value) + '</a>';
+								}
+							});
+						} else if (columnData.termId === 8255) {
+							// ENTRY_TYPE
+							columnsDef.push({
+								targets: columns.length - 1,
+								orderable: false,
+								render: function (data, type, full, meta) {
+									return '<a class="check-href edit-check' + full.entryId + '" href="javascript: void(0)" ' +
+										'onclick="showChangeEntryTypeModal(\'' + full.entryId + '\',\'' + data.value + '\',\'' + data.studyEntryPropertyId + '\')">' + EscapeHTML.escape(getCategoricalValue(data.value, columnData)) + '</a>';
+								}
+							});
+						} else if(columnData.termId < 0 && columnData.termId !== -6) {
```

Comment:
@mderamos @darla-leafnode 
This elseif block should be replaced by:
`						} else if (columnData.termId === -3) {
							//ACTIVE LOT
							columnsDef.push({
								targets: columns.length - 1,
								orderable: false,
								render: function (data, type, full, meta) {
									return full.lotCount;
								}
							});

						} else if (columnData.termId === -4) {
							// AVAILABLE
							columnsDef.push({
								targets: columns.length - 1,
								orderable: false,
								render: function (data, type, full, meta) {
									return full.availableBalance;
								}
							});

						} else if (columnData.termId === -5) {
							// UNIT
							columnsDef.push({
								targets: columns.length - 1,
								orderable: false,
								render: function (data, type, full, meta) {
									return full.unit;
								}
							});`
So we show the unit, available and lot columns and we cover https://ibplatform.atlassian.net/browse/IBP-3954 as part of this development, discussed during last planning session